### ISSUE TITLE
Add a more reliable way to enable/disable features

### ISSUE THREAD

**Issue by guillaumemolter** [bueltge/wordpress-multisite-enhancements#34]
Hi,

As I was disabling some features in the plugin using your recommended custom plugin mehtod, I realized that this method was really susceptible to break. It actually already wouldn't work if you follow the comments in the example plugin. Because you are using numeric keys to dequeue auto-load files and those files are auto-loaded alphabetically in the array, adding a new files (when adding new features) to the plugin would change the keys and therefor break all customizations.

I think the plugin should move to a more reliable way to do this, here are a few suggestions:
- Unset elements based on value using something like: 
```
if( false !== ( $key = array_search( $filename, $files ) ) ) {
    unset( $messages[$key] );
}
```
- Add a network setting page where user could enable/disable features.
- Add a dedicated filter in each class constructor allowing us to disable each feature (which could offer more flexibility when multiple features are bundled in one class).

Happy to contribute a patch based on what you think is the best solution.

**Comment by bueltge** [comment]
I'm feel with you, only the time in development was the point to realize that. I mean that an settings page is the best way for all users, there we currently have. A hook is shorter, no overhead for saving, controlling data, but easier to use. The biggest point to go in this direction. 
I would I  love if you help me in this context. Thanks a lot.

**Comment by bueltge** [comment]
Hi @guillaumemolter 
have you read my short answer, write from mobile? Do you have interest to work on this issue, solution?
Best regards.

**Comment by bueltge** [comment]
Thanks @cfoellmann 
simple change that helps in this context! 
Also updated the wiki page - https://github.com/bueltge/wordpress-multisite-enhancements/wiki/Remove-features

**Comment by bueltge** [comment]
#65 Should helps in this context and add a GUI for the de-activation of features.

**Comment by bueltge** [comment]
Solved with version 1.6.0
